internal/sign/job/manager.go

```diff
@@ -0,0 +1,133 @@
+package signjob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/sign"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const jobHashAnnotation = "kmm.node.kubernetes.io/last-hash"
+
+var errNoMatchingBuild = errors.New("no matching signing job")
+
+type signJobManager struct {
+	client client.Client
+	signer Signer
+	helper sign.Helper
+}
+
+func NewSignJobManager(client client.Client, signer Signer, helper sign.Helper) *signJobManager {
+	return &signJobManager{
+		client: client,
+		signer: signer,
+		helper: helper,
+	}
+}
+
+func labels(mod kmmv1beta1.Module, targetKernel string) map[string]string {
+	return map[string]string{
+		constants.ModuleNameLabel:    mod.Name,
+		constants.TargetKernelTarget: targetKernel,
+		constants.JobType:            "sign",
+	}
+}
+
+func (jbm *signJobManager) getJob(ctx context.Context, mod kmmv1beta1.Module, targetKernel string) (*batchv1.Job, error) {
+	jobList := batchv1.JobList{}
+
+	opts := []client.ListOption{
+		client.MatchingLabels(labels(mod, targetKernel)),
+		client.InNamespace(mod.Namespace),
+	}
+
+	if err := jbm.client.List(ctx, &jobList, opts...); err != nil {
+		return nil, fmt.Errorf("could not list jobs: %v", err)
+	}
+
+	if n := len(jobList.Items); n == 0 {
+		return nil, errNoMatchingBuild
+	} else if n > 1 {
+		return nil, fmt.Errorf("expected 0 or 1 job, got %d", n)
+	}
+
+	return &jobList.Items[0], nil
+}
+
+func (jbm *signJobManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string, previousImage string, targetImage string, pushImage bool) (sign.Result, error) {
```

Comment:
This function is almost exact copy of Sync function in internal/build/job/manager.go. Maybe we can move the bulk of the code to a common library, and both Sync functions will call it